### PR TITLE
Inspector review

### PR DESCRIPTION
**Pull request by XavierCLL**
* Fixes some issues and errors
* Little improvement in the current data raster provider
* Set initial structure for the others ee data providers
* Other adjustments

Tested with several examples on Linux, works perfectly

### CODE REVIEW

File: provider.py [hunk 1]
```diff
@@ -45,8 +49,8 @@ def providerKey(cls):
         return 'EE'
 
     @classmethod
-    def createProvider(cls, uri, providerOptions):
-        return EarthEngineRasterDataProvider(uri, providerOptions)
+    def createProvider(cls, uri, *args, **kwargs):
```
Comment (gena) [inline provider.py:52]: The arguments are incorrect, this is an overloaded method, see https://qgis.org/pyqgis/3.2/core/Provider/QgsProviderRegistry.html

File: utils.py [hunk 2]
```diff
@@ -128,8 +140,10 @@ def add_or_update_ee_layer(eeObject, visParams, name, shown, opacity):
         except:
             name = "untitled"
 
-    layer = add_or_update_ee_image_layer(image, name, shown, opacity)
+    image.ee_instance = type(eeObject)
```
Comment (gena) [inline utils.py:143]: Just a naming comment: ee_instance implies an object, but this is still a type, just a qgis_ee_type or a-like.

**Review by gena — COMMENTED** [review]
Looks good, should fix bugs in identify for non-image layers. I'm only not sure with the signature of the createProvider() method, there are no args and kwargs in QGIS API docs but providerOptions, so I'm not sure why this needs to be changed

File: provider.py (hunk 1 above)
Comment (XavierCLL) [reply provider.py]: yeah, right, when I tested it show me an error missing an argument I based on QgsProvideRregistry.py file that uses args and kwargs, but I check some test implementation and I found the missing argument is the flags, fixed in a40538f67723ef2d48a90be934152c91e8495d22 based on https://github.com/qgis/QGIS/blob/master/tests/src/python/provider_python.py#L198

**Review by gena — APPROVED** [review]

File: utils.py (hunk 2 above)
Comment (XavierCLL) [reply utils.py]: you're right, fixed in a8873156f89c8a6fe9f8d111173c1616b7c7a259

**Pull request merged**